src/password.cpp

```diff
@@ -0,0 +1,45 @@
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <wordexp.h>
+
+#include "password.h"
+
+const std::string read_password(const char* file) {
+    wordexp_t exp;
+    std::ifstream ifs;
+    std::string pass = "";
+
+    wordexp(file,&exp,0);
+    if (exp.we_wordc > 0) {
+        ifs.open(exp.we_wordv[0]);
+    }
+    wordfree(&exp);
+    if (!ifs.is_open()) {
+    } else {
+        std::getline(ifs, pass);
+    }
+    return pass;
+}
+
+const std::string eval_password(const char* cmd) {
+    char buffer[128];
+    std::string pass = "";
+
+    FILE* pipe = popen(cmd, "r");
+    if (!pipe) {
+    } else {
+        try {
+            while (!feof(pipe)) {
+                if (fgets(buffer, 128, pipe) != NULL) {
+                    buffer[strcspn(buffer, "\r\n")] = 0;
```

Comment:
OK, that's indeed pretty bad :/
Since I already switched to `utils::get_command_output`, I dropped this anyway and replaced it with methods from `std::string`.
It bugs me how hard and error-prone some easy tasks like "read until the first occurrence of (any) newline character" can be in C :(